app/src/main/java/io/zenandroid/onlinego/ui/screens/puzzle/PuzzleDirectoryState.kt

```diff
@@ -0,0 +1,11 @@
+package io.zenandroid.onlinego.ui.screens.puzzle
+
+import io.zenandroid.onlinego.data.model.local.VisitedPuzzleCollection
+import io.zenandroid.onlinego.data.model.local.PuzzleCollection
+import java.time.Instant
+
+data class PuzzleDirectoryState (
```

Comment:
Sadly, The way kotlin works with JVM, it cannot guarantee immutability of a Map. We have to add `@Immutable` here.